### PR TITLE
Dead OOC (Large Dsay)

### PR DESCRIPTION
**Pull request by Birdtalon**
# About the pull request

Some staff use YOOC (Yautja OOC) to talk to ghosts in a larger font which makes no sense. This adds a variant of dsay which is a larger font size for staff to talk to dchat if they need.

ps. also adds a space between the staff rank and the ckey ( because its annoying
dreamseeker_2024-03-19_19-15-01

# Explain why it's good for the game
# Testing Photographs and Procedure
<details>

<summary>Screenshots & Videos</summary>

Put screenshots and videos here with an empty line between the screenshots and the `<details>` tags.

</details>

# Changelog
:cl:
admin: DOOC - a larger deadsay for staff
fix: Adds a space between staff rank and ckey to match other dsay formatting.
/:cl:

### CODE REVIEW

**Comment by cm13-github** [comment]
This pull request has conflicts, please resolve those before we can evaluate the pull request.

File: tgui/packages/tgui-panel/styles/goon/chat-dark.scss [hunk 1]
```diff
@@ -301,10 +301,6 @@ em {
 .warningplain {
 }
 
-.deadsay {
-  color: #e2c1ff;
-}
-
```
Comment (Birdtalon) [inline tgui/packages/tgui-panel/styles/goon/chat-dark.scss:307]: This is a duplicated style definition

File: code/modules/admin/verbs/deadsay.dm [hunk 2]
```diff
@@ -47,3 +39,49 @@
 		return
 
 	dsay(msg)
+
+///Dsay but larger
+/client/proc/dooc(msg as text)
+	set category = "Admin.Events"
+	set name = "Dooc"
+	set hidden = TRUE
+	if(!src.admin_holder || !(admin_holder.rights & R_MOD))
```
Comment (Zonespace27) [inline code/modules/admin/verbs/deadsay.dm:48]:
```suggestion
	if(!admin_holder || !(admin_holder.rights & R_MOD))
```

File: code/modules/admin/verbs/deadsay.dm [hunk 3]
```diff
@@ -47,3 +39,49 @@
...
+		to_chat(src, "Only administrators may use this command.")
```
Comment (Zonespace27) [inline code/modules/admin/verbs/deadsay.dm:49]: add a SPAN_DANGER?

File: code/modules/admin/verbs/deadsay.dm [hunk 4]
```diff
@@ -47,3 +39,49 @@
...
+		return
+	if(prefs.muted & MUTE_DEADCHAT)
+		to_chat(src, SPAN_DANGER("You cannot send DOOC messages (muted)."))
+		return
+
+	if(!(prefs.toggles_chat & CHAT_DEAD))
+		to_chat(src, SPAN_DANGER("You have deadchat muted."))
+		return
+
+	if (src.handle_spam_prevention(msg,MUTE_DEADCHAT))
```
Comment (Zonespace27) [inline code/modules/admin/verbs/deadsay.dm:61]:
```suggestion
	if (handle_spam_prevention(msg, MUTE_DEADCHAT))
```

File: code/modules/admin/verbs/deadsay.dm [hunk 5]
```diff
@@ -47,3 +39,49 @@
...
+
+///Dsay but larger
+/client/proc/dooc(msg as text)
+	set category = "Admin.Events"
+	set name = "Dooc"
+	set hidden = TRUE
+	if(!src.admin_holder || !(admin_holder.rights & R_MOD))
+		to_chat(src, "Only administrators may use this command.")
+		return
+	if(!src.mob)
```
Comment (Zonespace27) [inline code/modules/admin/verbs/deadsay.dm:51]:
```suggestion
	if(!mob)
```

File: code/modules/admin/verbs/deadsay.dm [hunk 6]
```diff
@@ -47,3 +39,49 @@
...
+
+	if (!msg)
+		return
+
+	var/rendered = "<span class='dooc'><span class='prefix'>DEAD:</span> <span class='name'>[stafftype] ([src.key])</span> says, <span class='message'>\"[msg]\"</span></span>"
+
+	show_to_deadsay(rendered)
+
+/client/proc/show_to_deadsay(message)
+	for (var/mob/M in GLOB.player_list)
```
Comment (Zonespace27) [inline code/modules/admin/verbs/deadsay.dm:79]: as anything, no one letter vars, etc.

Comment (Zonespace27) [inline code/modules/admin/verbs/deadsay.dm:79]: also GLOB.observer_list may be better here

**Review by Zonespace27 — CHANGES_REQUESTED** [review]

**Comment by cm13-github** [comment]
Conflicts have been resolved. A maintainer will review the pull request shortly.

File: code/modules/admin/verbs/deadsay.dm [hunk 7]
```diff
@@ -47,3 +39,49 @@
...
+		return
+
+	var/rendered = "<span class='dooc'><span class='prefix'>DEAD:</span> <span class='name'>[stafftype] ([src.key])</span> says, <span class='message'>\"[msg]\"</span></span>"
+
+	show_to_deadsay(rendered)
+
+/client/proc/show_to_deadsay(message)
+	for (var/mob/mob in GLOB.observer_list)
+		if (istype(mob, /mob/new_player))
+			continue
```
Comment (Zonespace27) [inline code/modules/admin/verbs/deadsay.dm:81]:
```suggestion
```

File: code/modules/admin/verbs/deadsay.dm [hunk 8]
```diff
@@ -47,3 +39,47 @@
...
+
+	if (!msg)
+		return
+
+	var/rendered = "<span class='dooc'><span class='prefix'>DEAD:</span> <span class='name'>[stafftype] ([src.key])</span> says, <span class='message'>\"[msg]\"</span></span>"
+
+	show_to_deadsay(rendered)
+
+/client/proc/show_to_deadsay(message)
+	for (var/mob/mob in GLOB.observer_list)
```
Comment (Drulikar) [inline code/modules/admin/verbs/deadsay.dm:79]: Did you mean to change this to observer? If so the admin check below on 81 I think is moot.

File: code/modules/admin/verbs/deadsay.dm [hunk 9]
```diff
@@ -47,3 +39,47 @@
...
+
+		if(mob.client && mob.client.admin_holder && (mob.client.admin_holder.rights & R_MOD) && mob.client.prefs && (mob.client.prefs.toggles_chat & CHAT_DEAD)) // show the message to admins who have deadchat toggled on
+			mob.show_message(message, SHOW_MESSAGE_AUDIBLE)
+
+		else if((mob.stat == DEAD || isobserver(mob)) && mob && mob.client && mob.client.prefs && (mob.client.prefs.toggles_chat & CHAT_DEAD)) // show the message to regular ghosts who have deadchat toggled on
```
Comment (Drulikar) [inline code/modules/admin/verbs/deadsay.dm:84]:
```suggestion
		else if((mob.stat == DEAD || isobserver(mob)) && mob.client?.prefs && (mob.client.prefs.toggles_chat & CHAT_DEAD)) // show the message to regular ghosts who have deadchat toggled on
```

File: code/modules/admin/verbs/deadsay.dm [hunk 10]
```diff
@@ -47,3 +39,47 @@
...
+		return
+
+	var/rendered = "<span class='dooc'><span class='prefix'>DEAD:</span> <span class='name'>[stafftype] ([src.key])</span> says, <span class='message'>\"[msg]\"</span></span>"
+
+	show_to_deadsay(rendered)
+
+/client/proc/show_to_deadsay(message)
+	for (var/mob/mob in GLOB.observer_list)
+
+		if(mob.client && mob.client.admin_holder && (mob.client.admin_holder.rights & R_MOD) && mob.client.prefs && (mob.client.prefs.toggles_chat & CHAT_DEAD)) // show the message to admins who have deadchat toggled on
```
Comment (Drulikar) [inline code/modules/admin/verbs/deadsay.dm:81]:
```suggestion
		if(mob.client?.admin_holder && (mob.client.admin_holder.rights & R_MOD) && mob.client.prefs && (mob.client.prefs.toggles_chat & CHAT_DEAD)) // show the message to admins who have deadchat toggled on
```

**Review by Drulikar — CHANGES_REQUESTED** [review]

**Pull request closed without merging**